src/events.c

```diff
@@ -402,9 +405,16 @@ void al_wait_for_event(ALLEGRO_EVENT_QUEUE *queue, ALLEGRO_EVENT *ret_event)
 
    _al_mutex_lock(&queue->mutex);
    {
+      #ifdef __EMSCRIPTEN__
+      while (is_event_queue_empty(queue)) {
```

Comment:
I don't suppose al_wait_cond_until would just work here (no idea if it's implemented)? Do condvars even work on emscripten?

Reply:
It depends. You can compile emscripten with pthreads enabled: https://emscripten.org/docs/porting/pthreads.html

I just couldn't get it to work myself, but I can try again. Without pthreads none of the threading/locking related functions do anything at all so al_get_next_event() just instantly deadlocks the browser in that case.

Reply:
Hm, I tried with -pthread but then I hit: https://github.com/emscripten-ports/SDL2/issues/110

I'm using the SDL2 that comes prepackaged with emscripten right now so not really sure how to apply those fixes to try it. I'll see if I can fix the audio issue before I attempt that.